release-notes.txt

```diff
@@ -1,3 +1,566 @@
+RIOT-2025.10 "OCT31 == DEC25" - Release Notes
+============================
+RIOT is a multi-threading operating system which enables soft real-time
+capabilities and comes with support for a range of devices that are typically
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2025.10 release includes:
+
+ - The new XIPFS filesystem module allows executing code from memory mapped flash on
+   supported platforms.
+
+ - Support for the ESP32-C6 RISC-V WiFi MCU was added.
+
+ - NanoCoAP server will no longer copy requests to an internal buffer for processing, instead
+   it operates directly on the network buffer, eliminating the case of 'too large' requests.
+
```

Comment:
While it was mentioned in last release when the foundation was set up, in this release nearly all of the guides were migrated, adapted or rewritten to the new guide site, including proper structuring etc.. I'd say around 95% of the guide migration happened within this release.

#21741 also included a neat parser so you can now also read all the changelogs easily at https://guide.riot-os.org/changelog/ which feels cool to mention in the current anti-github climate.